connaisseur/tests/test_flask_server.py

```diff
@@ -193,246 +77,94 @@ def test_readyz(
-    mock_request_data["request"]["object"]["spec"]["containers"][0]["image"] = image
-    response = client.post("/mutate", json=mock_request_data)
-    assert response.status_code == 200
-    assert response.is_json
-
-    admission_response = response.get_json()["response"]
-
-    assert admission_response["allowed"] == allowed
-
-    if allowed:
-        assert admission_response["status"]["code"] == 202
-        assert not "message" in admission_response["status"]
-        mock_send_alerts.assert_has_calls(
-            [mocker.call(mock_request_data, admitted=True)]
-        )
-    else:
-        assert admission_response["status"]["code"] == 403
-        image = (
-            image
-            if image.startswith("docker.io/")
-            else "{}{}".format("docker.io/", image)
-        )
-        detection_mode_string = (
-            " (not denied due to DETECTION_MODE)" if detection == "1" else ""
-        )
-        expected_message = (
-            'could not find signed digest for image "{}" in trust data.{}'.format(
-                image, detection_mode_string
-            )
-        )
-        assert admission_response["status"]["message"] == expected_message
-        mock_send_alerts.assert_has_calls(
-            [mocker.call(mock_request_data, admitted=False, reason=expected_message)]
-        )
-
-
-@pytest.mark.parametrize("ad_request_filename", [("ad_request_deployments")])
-def test_alert_sending_error_handler(
-    mocker,
-    requests_mock,
-    mock_config,
-    mock_mutate,
-    mock_policy_verify,
-    mock_env_vars,
-    monkeypatch,
-    ad_request_filename,
-):
-    monkeypatch.setenv("DETECTION_MODE", "0")
-
-    requests_mock.post(
-        "https://www.mocked_test_connaisseur_url.xyz",
-        status_code=401,
-    )
-    try:
-        mock_response = requests.post(
-            "https://www.mocked_test_connaisseur_url.xyz", {}, {}
-        )
-        mock_response.raise_for_status()
-    except HTTPError as err:
-        http_error = err
-    mocker.patch("connaisseur.flask_server.admit", return_value=True)
-    mock_send_alerts = mocker.patch(
-        "connaisseur.flask_server.send_alerts",
-        side_effect=AlertSendingError(http_error),
-    )
-    client = pytest.fs.APP.test_client()
-    mock_request_data = get_file_json(f"tests/data/{ad_request_filename}.json")
-    response = client.post("/mutate", json=mock_request_data)
-    mock_send_alerts.assert_has_calls([mocker.call(mock_request_data, admitted=True)])
-    assert response.status_code == 500
-    assert (
-        response.get_data().decode()
-        == "Alert could not be sent. Check the logs for more details!"
-    )
-
-
-@pytest.mark.parametrize("ad_request_filename", [("ad_request_deployments")])
-def test_configuration_error_handler(
-    mocker,
-    mock_mutate,
-    mock_policy_verify,
-    mock_env_vars,
-    mock_config,
-    monkeypatch,
-    ad_request_filename,
+def test_admit(
```

Comment:
I don't see tests for the flask error handlers, am I wrong or have they been lost in the process? :-) 

Same holds for the tests whether the alert sending has been called or not as far as I see it

Reply:
i added the `test_error_handler` tests to `test_flask_server.py`